### PR TITLE
JBPM-5163: Remove old Dashbuilder from Workbench

### PR DESCRIPTION
**Pull request by dgutierr**
Changes required in order to integrate the new dashbuilder CMS features into the workbench. 

Video preview of the new features => https://www.youtube.com/watch?v=SSrYP3meBT8

See also:
https://www.youtube.com/watch?v=51ETHDJsjtA
https://www.youtube.com/watch?v=eItjBB4j0CY

Those changes consists basically of replacing the old dashbuilder tooling by the new dashbuilder CMS perspective => https://issues.jboss.org/browse/DASHBUILDE-166  . The feature itself has already been  gone through a complete QE testing and bug fixing cycle.

@wmedvede Can you please take a look and approve this PR?  Merge deadline is Wednesday EOD.

PS: depends on https://github.com/kiegroup/kie-wb-common/pull/764

### CODE REVIEW

**Comment by dgutierr** [comment]
@psiroky Build is failing. Not related to test though. Any clues?

**Comment by psiroky** [comment]
The error is there in build log:
```
   Tracing compile failure path for type 'org.kie.workbench.common.workbench.client.menu.DefaultWorkbenchFeaturesMenusHelper'
13:57:13 [INFO]       [ERROR] Errors in 'jar:file:/home/jenkins/.m2/repository/org/kie/workbench/screens/kie-wb-common-workbench-client/7.0.0-SNAPSHOT/kie-wb-common-workbench-client-7.0.0-SNAPSHOT-sources.jar!/org/kie/workbench/common/workbench/client/menu/DefaultWorkbenchFeaturesMenusHelper.java'
13:57:13 [INFO]          [ERROR] Line 279: No source code is available for type org.dashbuilder.navigation.NavTree; did you forget to inherit a required module?
13:57:13 [INFO]          [ERROR] Line 281: No source code is available for type org.dashbuilder.navigation.NavItem; did you forget to inherit a required module?
13:57:13 [INFO]          [ERROR] Line 284: No source code is available for type org.dashbuilder.navigation.NavDivider; did you forget to inherit a required module?
13:57:13 [INFO]          [ERROR] Line 294: No source code is available for type org.dashbuilder.navigation.NavGroup; did you forget to inherit a required module?
13:57:13 [INFO]          [ERROR] Line 299: No source code is available for type org.dashbuilder.navigation.workbench.NavWorkbenchCtx; did you forget to inherit a required module?
13:57:15 [INFO]    [ERROR] Aborting compile due to errors in some input files
```

**Comment by dgutierr** [comment]
I've identified the cause. Which I do not understand why it wasn't failing while building locally.

@psiroky Thanks for helping.

**Comment by dgutierr** [comment]
jenkins retest this

**Comment by wmedvede** [comment]
Hello @dgutierr I'm getting the same error above I guess when trying to build you changes locally.
(I've built both dashbuilder and kie-wb-commons with the changes of your PR prior to building kie-wb-distributions locally)

Would you mind take a look?


[INFO]       [ERROR] Errors in 'jar:file:/home/wmedvede/.m2/repository/org/kie/workbench/screens/kie-wb-common-workbench-client/7.0.0-SNAPSHOT/kie-wb-common-workbench-client-7.0.0-SNAPSHOT-sources.jar!/org/kie/workbench/common/workbench/client/menu/DefaultWorkbenchFeaturesMenusHelper.java'
[INFO]          [ERROR] Line 279: No source code is available for type org.dashbuilder.navigation.NavTree; did you forget to inherit a required module?
[INFO]          [ERROR] Line 281: No source code is available for type org.dashbuilder.navigation.NavItem; did you forget to inherit a required module?
[INFO]          [ERROR] Line 284: No source code is available for type org.dashbuilder.navigation.NavDivider; did you forget to inherit a required module?
[INFO]          [ERROR] Line 294: No source code is available for type org.dashbuilder.navigation.NavGroup; did you forget to inherit a required module?
[INFO]          [ERROR] Line 299: No source code is available for type org.dashbuilder.navigation.workbench.NavWorkbenchCtx; did you forget to inherit a required module?
[INFO]    [ERROR] Aborting compile due to errors in some input files
[I

**Comment by dgutierr** [comment]
jenkins retest this

**Comment by dgutierr** [comment]
jenkins retest this

**Comment by wmedvede** [comment]
@dgutierr  The integration seems to be ok and the build errors are now fixed. It was also possible to run the feature but only in kie-wb, when I deploy kie-drools-wb and login with the same user and rights that I did in kie-wb the feature is not available. Is this ok?

**Comment by dgutierr** [comment]
@wmedvede Yes, good point Walter.  The ability to create new dashboards at runtime and attach them to the menu bar it is jBPM specific, so no need, at the moment,  for including this feature in the drools deliverables.

**Comment by dgutierr** [comment]
@jhrcek There are 3 selenium tests failing. I do not know the reason TBH. Can you take a look please?

**Comment by jhrcek** [comment]
Failing test has:
```
Unable to locate element: {"method":"link text","selector":"Artifact repository"}
```
and the [screenshot](https://kie-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/kie-wb-distributions-pullrequests/607/artifact/kie-wb-distributions/kie-wb-tests/kie-wb-tests-gui/target/screenshots/LoadAllPerspectivesIntegrationTest_allPerspectivesCanBeLoaded.png) from failing test shows "Artifact Repository" with uppercase R.

I see this PR changed some I18n constants. If you update the constants [here](https://github.com/kiegroup/kie-wb-distributions/blob/master/kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/Persp.java#L42) accordingly, the tests should pass?

**Comment by wmedvede** [comment]
@dgutierr Ok, thanks for the explanation. 
+1 from my side when the i18n constants  reported by @jhrcek are fixed.

Thanks.

**Comment by dgutierr** [comment]
@jhrcek Great. I have forced them to match. Thanx!

File: kie-wb-parent/kie-wb-webapp/pom.xml [hunk 1]
```diff
@@ -1216,6 +1224,18 @@
...
+      <groupId>org.dashbuilder</groupId>
+      <artifactId>dashbuilder-navigation-client</artifactId>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.dashbuilder</groupId>
+      <artifactId>dashbuilder-cms-client</artifactId>
+      <scope>provided</scope>
+    </dependency>
```
Comment (psiroky) [inline kie-wb-parent/kie-wb-webapp/pom.xml]: Please remove these `provided` client deps. They are not needed. I am preparing a PR which will remove them, but lets just add more.

**Comment by cristianonicolai** [comment]
@dgutierr could you also update the jbpm-wb-showcase to use the updated dashbuilder?

**Comment by dgutierr** [comment]
@cristianonicolai Sure! I will

**Pull request merged**